### PR TITLE
ARROW-12729: [R] Fix length method for Table, RecordBatch

### PR DESCRIPTION
**Pull request by thisisnic**

### CODE REVIEW

**Review by ianmcook — APPROVED** [review]
LGTM!

**Comment by ianmcook** [comment]
Oh, actually one small requested change please: both `Table` and `RecordBatch` inherit the virtual class `ArrowTabular`, so whenever we want to define a method for both `Table` and `RecordBatch` and the implementation of it is exactly the same, we should instead define it on `ArrowTabular`. Could you please change this to use that? The tests can stay the same.

**Comment by thisisnic** [comment]
> Oh, actually one small requested change please: both `Table` and `RecordBatch` inherit the virtual class `ArrowTabular`, so whenever we want to define a method for both `Table` and `RecordBatch` and the implementation of it is exactly the same, we should instead define it on `ArrowTabular`. Could you please change this to use that? The tests can stay the same.

Good point, completely missed that, now done!

**Review by romainfrancois — APPROVED** [review]

File: r/R/table.R [hunk 1]
```diff
@@ -184,3 +184,6 @@ Table$create <- function(..., schema = NULL) {
 
 #' @export
 names.Table <- function(x) x$ColumnNames()
+
+#' @export
+length.ArrowTabular <- function(x) x$num_columns
```
Comment (ianmcook) [inline r/R/table.R:189]: Could you please move this to `arrow-tabular.R`? Thanks!

Comment (thisisnic) [reply r/R/table.R]: Done!

**Pull request closed without merging**